tests/gdx-tests/src/com/badlogic/gdx/tests/g3d/shadows/utils/DirectionalAnalyzer.java

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.tests.g3d.shadows.utils;
+
+import com.badlogic.gdx.graphics.Camera;
+import com.badlogic.gdx.graphics.g3d.environment.BaseLight;
+import com.badlogic.gdx.math.Frustum;
+import com.badlogic.gdx.math.Vector3;
+
+/** Directional Analyzer compute the camera's properties needed by a directional light
+ * @author realitix */
+public interface DirectionalAnalyzer {
+	/** Compute the good orthographicCamera dimension based on the frustum. Be careful, direction must be normalized.
+	 * @param light Current light
+	 * @param frustum Frustum of the main camera
+	 * @param direction Direction of the directional light
+	 * @param out Camera modified ( still needs to call update() )
+	 * @return Camera The camera for chaining */
+	public Camera analyze (BaseLight light, Frustum frustum, Vector3 direction, Camera out);
```

Comment:
Not really a big thing, but I have a feeling it is better to pass in the actual `Camera` instead of the `Frustum` (both from a user which might not even know where to get this frustum from, as well as someone who wants to make his own implementation).
If it is only intended for directional light then the `light` argument might as well be a `DirectionalLight` so the `direction` argument isn't needed. From what I've seen in the implementation, the `light` argument is never used.